cscs-checks/apps/amber/amber_check.py

```diff
@@ -197,3 +144,10 @@ def set_num_tasks_cray_shasta(self):
                 # need to decrease to just 8 nodes
                 self.num_nodes = 8
             self.num_tasks = self.num_nodes * self.num_tasks_per_node
+
+    @run_after('setup')
+    def set_perf_reference(self):
+        if self.scale == 'small':
+            self.reference = REFERENCE_CPU_PERFORMANCE_SMALL
```

Comment:
I'd vote for a single multi-level dictionary of references, so that we can get the tuple that we want with a single line here.

Reply:
I agree, it would make things easier. But then we would have to change the `set_perf_patterns` function for the new kind of dictionaries, respectively we would have to add a new artificial level of keys in `REFERENCE_GPU_PERFORMANCE` and a new parameter (`scale`) for the GPU test. I think the current version is more natural, don't you think?